build_sysext

```diff
@@ -0,0 +1,103 @@
+#!/bin/bash
+
+# Copyright (c) 2023 Flatcar Linux. All rights reserved.
+# Use of this source code is governed by a BSD-style license that can be
+# found in the LICENSE file.
+
+# Script to generate sysext. See systemd-sysext(8). Prerequisite that you've run build_packages and build_image
+
+SCRIPT_ROOT=$(dirname "$(readlink -f "$0")")
+. "${SCRIPT_ROOT}/common.sh" || exit 1
+
+# Script must run inside the chroot
+assert_inside_chroot
+
+DEFAULT_GROUP=developer
+DEFAULT_DEVCONTAINER_BINHOST="${SETTING_BINPKG_SERVER_PROD}"
+
+. "${BUILD_LIBRARY_DIR}/toolchain_util.sh" || exit 1
+. "${BUILD_LIBRARY_DIR}/build_image_util.sh" || exit 1
+. "${BUILD_LIBRARY_DIR}/vm_image_util.sh" || exit 1
+. "${BUILD_LIBRARY_DIR}/cros_vm_constants.sh" || exit 1
+
+DEFINE_string board "${DEFAULT_BOARD}" \
+  "Board for which the sysext was built"
+
+# include upload options
+. "${BUILD_LIBRARY_DIR}/release_util.sh" || exit 1
+
+FLAGS_HELP="USAGE: build_sysext [flags] [package to build].
+This script is used to build a Flatcar sysext. 
+
+Examples:
+
+build_sysext --board=amd64 python - builds python sysext for amd64 architecture.
+
+"
+
+show_help_if_requested "$@"
+
+# Parse command line
+FLAGS "$@" || exit 1
+
+eval set -- "${FLAGS_ARGV}"
+
+# N.B.  Ordering matters for some of the libraries below, because
+# some of the files contain initialization used by later files.
+. "${BUILD_LIBRARY_DIR}/toolchain_util.sh" || exit 1
+. "${BUILD_LIBRARY_DIR}/board_options.sh" || exit 1
+. "${BUILD_LIBRARY_DIR}/build_image_util.sh" || exit 1
+. "${BUILD_LIBRARY_DIR}/prod_image_util.sh" || exit 1
+. "${BUILD_LIBRARY_DIR}/dev_container_util.sh" || exit 1
+. "${BUILD_LIBRARY_DIR}/test_image_content.sh" || exit 1
+. "${BUILD_LIBRARY_DIR}/torcx_manifest.sh" || exit 1
+. "${BUILD_LIBRARY_DIR}/vm_image_util.sh" || exit 1
+
+BUILD_DIR="${DEFAULT_BUILD_ROOT}/images/${BOARD}/${IMAGE_SUBDIR}"
+
+
+cleanup() {
+    # Unmount the directory here
+    umount "${BUILD_DIR}/fs-root" "${BUILD_DIR}/install-root" "${BUILD_DIR}/workdir" || exit 1
+}
+
+
+# export SYSEXTNAME=$2
+
+ARCH=[[ "$BUILD_DIR" == *arm64-usr* ]] && ARCH=arm64 || [[ "$BUILD_DIR" == *amd64-usr* ]] && ARCH=x86-64
+
+source /etc/os-release
+
+
+unmount "${BUILD_DIR}/fs-root" "${BUILD_DIR}/install-root" "${BUILD_DIR}/workdir" || exit 1
+
+
+# # with my 1 line patch before the rms in build_library/prod_image_util.sh (mksquashfs "${root_fs_dir}" "${BUILD_DIR}/configroot" "${BUILD_DIR}"/flatcar_production_image_sysext.sqsh)
+
+VERSION_BOARD=$(awk -F= '$1=="FLATCAR_VERSION_ID" { print $2 ;}' "${BUILD_DIR}/version.txt")
+
+if [ "$VERSION_BOARD" == "$VERSION_ID" ]; then
+    sudo mkdir "${BUILD_DIR}/fs-root"
+    sudo mount -rt squashfs -o loop,nodev,noexec "${BUILD_DIR}/flatcar_production_image_sysext.sqsh" "${BUILD_DIR}/fs-root"
```

Comment:
At this point we should check the `/usr/lib/os-release` file for the correct version.

Reply:
Oh, I get Kai's point - we could mount the squashfs and extract `FLATCAR_VERSION_ID` from there (`...fs-root/usr/lib/os-release`) instead of using `version.txt` from `${BUILDDIR}`. 
`version.txt` is not guaranteed to be in the build directory - for instance if a user downloads the sysext squasfs of a release to build a custom sysext instead of running a full build themselves.

Reply:
The field to extract from in the `os-release` file is `VERSION` and this should match the version the SDK has built/would build.
The field `VERSION_ID` for the extension release file is the `VERSION_ID` field from the `os-release` file (that's missing the build ID but that's Flatcar-specific and not supported for matching - means there are corner cases where the sysext isn't rejected but we can't prevent this).